mapper/yml2image.py

```diff
@@ -41,21 +66,33 @@ def legal_env_spec(es):
                 report_and_exit(f'Problem with deployment spec: {env} does not specific tag or branch pattern')
     return False
 
-def validate(spec):
-    name = spec.get('name') or report_and_exit("Problem with deployment spec: couldn't find image.name")
 
-    deployments = spec.get('deployments')
-    if not deployments:
-        report_and_exit("Problem with deployment spec: couldn't find deployments")
-    if not isinstance(deployments, list):
-        report_and_exit("Problem with deployment spec: deployments should be an ordered list of environments")
-    if not all( legal_env_spec(es) for es in deployments):
-        report_and_exit("Problem with deployment spec: each environment spec should have a tag or branch specified")
-    return name
+def find_ref(spec, ref):
+    target = re.sub('refs/(tags|heads)/','', ref)
+    for d in spec.get('deployments'):
+        if d.get('tag'):
+            pattern = d.pop('tag')
+        elif d.get('branch'):
+            pattern = d.pop('branch')
+        if (pattern):
+            pattern = pattern.replace("{ver}", "[0-9][0-9\\.]*")
+            # print(f'pattern:{pattern}')
+            try: 
+                if re.match(pattern, target):
```

Comment:
Why the `match` instead of `fullMatch`? That explains the comment in the readme about ordering but why the difference?

Reply:
I'll change it back.